pkg/volume/csi/csi_plugin.go

```diff
@@ -846,6 +852,43 @@ func (p *csiPlugin) getVolumeLifecycleMode(spec *volume.Spec) (storage.VolumeLif
 	return storage.VolumeLifecyclePersistent, nil
 }
 
+// getSupportsFSGroup returns  if the CSI driver supports  supports a volume in the given mode.
+// An error indicates that it isn't supported and explains why.
+func (p *csiPlugin) getSupportsFSGroup(driver string) (storage.SupportsFSGroup, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIVolumeSupportsFSGroup) {
+		// feature is disabled, default to Evaluate
+		return storage.EvaluateSupportsFSGroup, nil
+	}
+
+	// Retrieve CSIDriver. It's not an error if that isn't
+	// possible (we don't have the lister if CSIDriverRegistry is
+	// disabled) or the driver isn't found (CSIDriver is
+	// optional)
+	var csiDriver *storage.CSIDriver
+	if p.csiDriverLister != nil {
+		kletHost, ok := p.host.(volume.KubeletVolumeHost)
+		if ok {
+			if err := kletHost.WaitForCacheSync(); err != nil {
```

Comment:
Hmm I see this being called elsewhere, but I wonder if this is really the right place to wait for the initial sync. It would probably be better to wait during some volumemanager initialization. It may be good to open a separate issue to try to clean this up.

Reply:
I'm going to leave this section untouched for now, and we can open a separate issue to adjust the csiDriver initialization.